lms/djangoapps/discussion/views.py

```diff
@@ -557,7 +604,14 @@ def followed_threads(request, course_key, user_id):
         query_params['num_pages'] = paginated_results.num_pages
         user_info = cc.User.from_django_user(request.user).to_dict()
 
-        with newrelic.agent.FunctionTrace(nr_transaction, "get_metadata_for_threads"):
+        if newrelic:
+            with newrelic.agent.FunctionTrace(nr_transaction, "get_metadata_for_threads"):
+                annotated_content_info = utils.get_metadata_for_threads(
+                    course_key,
+                    paginated_results.collection,
+                    request.user, user_info
+                )
+        else:
             annotated_content_info = utils.get_metadata_for_threads(
```

Comment:
Mainly I didn't understand the 'with' keyword well enough. Hopefully what I refactored into works, or is on the right track :P